types/src/transaction/action.rs

```diff
@@ -475,6 +475,34 @@ impl Action {
             _ => None,
         }
     }
+
+    pub fn key_allowance(&self) -> KeyAllowance {
+        match self {
+            Action::SetRegularKey {
+                ..
+            } => KeyAllowance {
+                sole_master_key: true,
+                master_key: true,
+            },
+            _ => Default::default(),
+        }
+    }
+}
+
+pub struct KeyAllowance {
+    // Default key that doesn't have regular key.
+    pub sole_master_key: bool,
+    // If any regular key is set.
+    pub master_key: bool,
```

Comment:
Is `master_key` `true` when the transaction doesn't allow the `regular key`?
It seems that the name doesn't match the behavior.

Reply:
Okay. I misread.
Is it `true` when the transaction with master key signed is not allowed if the account has a regular key?
I think the name is too short to describe the behavior.

Reply:
Yes. That's why I've made `sole_master_key`. I thought it would help to distinguish a master key that has a regular key with a key that doesn't. I'll make them more descriptive.

Reply:
I categorized them into this. It should compatible with `is_master_key_allowed`
```
+- Sole master key
| +- Master key
| | +- Regular key
O O O => is_master_key_allowed()
O X O => !is_master_key_allowed()
O O X => SelfNomination
```